### PR TITLE
[SPARK-34952][SQL][FOLLOWUP] Simplify JDBC aggregate pushdown

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

This is a followup of https://github.com/apache/spark/pull/33352 , to simplify the JDBC aggregate pushdown:
1. We should get the schema of the aggregate query by asking the JDBC server, instead of calculating it by ourselves. This can simplify the code a lot, and is also more robust: the data type of SUM may vary in different databases, it's fragile to assume they are always the same as Spark.
2. because of 1, now we can remove the `dataType` property from the public `Sum` expression.

This PR also contains some small improvements:
1. Spark should deduplicate the aggregate expressions before pushing them down.
2. Improve the `toString` of public aggregate expressions to make them more SQL.

### Why are the changes needed?

code and API simplification

### Does this PR introduce _any_ user-facing change?

this API is not released yet.

### How was this patch tested?

existing tests

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @huaxingao @viirya

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2ScanRelationPushDown.scala [hunk 1]
```diff
@@ -76,9 +78,18 @@ object V2ScanRelationPushDown extends Rule[LogicalPlan] with PredicateHelper {
...
           sHolder.builder match {
             case _: SupportsPushDownAggregates =>
+              val aggExprToOutputOrdinal = mutable.HashMap.empty[Expression, Int]
+              var ordinal = 0
               val aggregates = resultExpressions.flatMap { expr =>
                 expr.collect {
-                  case agg: AggregateExpression => agg
+                  // Do not push down duplicated aggregate expressions. For example,
+                  // `SELECT max(a) + 1, max(a) + 2 FROM ...`, we should only push down one
+                  // `sum(a)` to the data source.
```
Comment (huaxingao) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2ScanRelationPushDown.scala:87]: nit: `max(a)`?

**Comment by huaxingao** [comment]
java linter failed because of `Unused import - org.apache.spark.sql.types.DataType` in `Sum`.

Thanks a lot for improving the code! Really appreciate your help!

**Review by huaxingao — APPROVED** [review]

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCScanBuilder.scala [hunk 2]
```diff
@@ -51,56 +57,53 @@ case class JDBCScanBuilder(
...
+    val jdbcOptionsWithAggQuery = new JDBCOptions(
+      jdbcOptions.parameters
+        - JDBCOptions.JDBC_TABLE_NAME
+        - JDBCOptions.JDBC_PARTITION_COLUMN
+        - JDBCOptions.JDBC_NUM_PARTITIONS
+        - JDBCOptions.JDBC_LOWER_BOUND
+        - JDBCOptions.JDBC_UPPER_BOUND +
+        (JDBCOptions.JDBC_QUERY_STRING -> aggQuery))
+    try {
+      finalSchema = JDBCRDD.resolveTable(jdbcOptionsWithAggQuery)
```
Comment (viirya) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCScanBuilder.scala:98]: Oh, this is a good change.

**Review by viirya — APPROVED** [review]

**Review by viirya — APPROVED** [review]

**Comment by viirya** [comment]
CI pending.

**Comment by viirya** [comment]
Thanks. Merging master/3.2.

**Pull request closed without merging**